Dapps modals

varunguleriaCodes opened the pull request:
## Description and Related Issue(s)

This is a pr for https://github.com/blockscout/frontend/issues/2454
### Proposed Changes

Added a popup for an external dapp as specified in  https://github.com/blockscout/frontend/issues/2454


### Additional Information
*[Include any additional information, context, or screenshots that may be helpful for reviewers.]*
Pc View
![Screenshot from 2025-01-02 00-28-24](https://github.com/user-attachments/assets/23ef02e0-ac30-48dc-a3ea-43524e86527f)

Mobile View
![Screenshot from 2025-01-02 00-28-58](https://github.com/user-attachments/assets/b48533a9-442a-48c2-a893-26a11c8beab9)


## Checklist for PR author
- [Done ] I have tested these changes locally.

isstuev commented:
Hi @varunguleriaCodes 
Thank you for contributing.
Actually we already have dapp modal implemented on the dapps page, so no need to implement it again.
ui/marketplace/MarketplaceAppModal.tsx
<img width="1244" alt="image" src="https://github.com/user-attachments/assets/0b791ca2-a85f-440c-b2e5-135618afef6d" />

All we need to change in this task is to open /apps page (with a given dapp modal opened) instead of dapp link (only for external dapps).

isstuev reviewed (changes requested):
please use an existing page and component

varunguleriaCodes commented:
Hi @isstuev , got it working on suggested changes

isstuev commented on ui/marketplace/useMarketplace.tsx:
```diff
@@ -106,6 +106,23 @@ export default function useMarketplace() {
+    if (selectedAppId) {
+      setSelectedAppId(selectedAppId);
+      setIsAppInfoModalOpen(true);
+      const newQuery = { ...router.query };
```
please use frontend/lib/router/removeQueryParam.ts

isstuev commented on ui/snippets/searchBar/SearchBarSuggest/SearchBarSuggestApp.tsx:
```diff
@@ -1,5 +1,6 @@
 import { Image, Flex, Text, useColorModeValue } from '@chakra-ui/react';
```
We also have Search Results page where apps can also be displayed, I suppose we need to implement the same logic there
frontend/ui/searchResults/SearchResultTableItem.tsx
frontend/ui/searchResults/SearchResultListItem.tsx

varunguleriaCodes commented:
Hi @isstuev , I have added the changes, also changed useEffect in rating components, as when I was redirecting in a new tab to /apps this error showed "maximum update depth error reached"

tom2drum commented on ui/snippets/searchBar/SearchBarSuggest/SearchBarSuggestApp.tsx:
```diff
@@ -96,9 +96,22 @@ const SearchBarSuggestApp = ({ data, isMobile, searchTerm, onClick }: Props) =>
+          },
+        }}
+        passHref
+        shallow={ router.pathname === '/apps' }
```
@varunguleriaCodes, the test has failed again. I believe this is because you are using `router` directly from the import instead of the router instance obtained through the `useRouter` hook.

varunguleriaCodes replied:
@tom2drum , updated the import

tom2drum replied:
All good now.

isstuev commented on ui/searchResults/SearchResultListItem.tsx:
```diff
@@ -146,7 +146,14 @@ const SearchResultListItem = ({ data, searchTerm, isLoading, addressFormat }: Pr
             { data.app.external ? (
               <LinkExternal
-                href={ data.app.url }
+                href={
```
it's not an external link anymore, need to change the link component as well
